Commit c225a6cfce24bd7b14add5c91822cfe7a78c1bcb breaks a pool with a memory file

The following code works before c225a6cfce24bd7b14add5c91822cfe7a78c1bcb, but does not work afterward:

```go
package sqlite

import (
	"context"
	"errors"
	"testing"

	"crawshaw.io/sqlite"
	"crawshaw.io/sqlite/sqliteutil"
)

const schema = `CREATE TABLE IF NOT EXISTS test (col VARCHAR(255));`

func TestSelect(t *testing.T) {
	pool, cleanup := newMemoryPool(t)
	defer cleanup()

	values, err := selectStar(context.Background(), pool)
	if err != nil {
		t.Fatal(err)
	}

	t.Log(values)
}

func selectStar(ctx context.Context, pool *sqlite.Pool) ([]string, error) {
	conn := pool.Get(ctx.Done())
	if conn == nil {
		return nil, errors.New("failed to get connection")
	}
	defer pool.Put(conn)

	query := "SELECT * FROM test;"
	stmt, err := conn.Prepare(query) // This returns an error
	if err != nil {
		return nil, err
	}
	values := []string{}
	for {
		if hasRow, err := stmt.Step(); err != nil {
			return nil, err
		} else if !hasRow {
			break
		}
		value := stmt.GetText("value")
		values = append(values, value)
	}
	return values, nil
}

func newMemoryPool(t *testing.T) (*sqlite.Pool, func()) {
	pool, err := sqlite.Open("file:memdb?mode=memory", 0, 10)
	if err != nil {
		t.Fatal(err)
		return nil, func() {}
	}
	if err := setup(pool); err != nil {
		t.Fatal(err)
		return nil, func() {}
	}
	return pool, func() { pool.Close() }
}

func setup(pool *sqlite.Pool) error {
	conn := pool.Get(context.Background().Done())
	if conn == nil {
		return errors.New("failed to get connection")
	}
	defer pool.Put(conn)
	return sqliteutil.ExecScript(conn, schema)
}
```

When I run it, I get the following error:

```plain
$ go test -v .
# crawshaw.io/sqlite
cgo-generated-wrappers:7:13: warning: built-in function ‘free’ declared as non-function
=== RUN   TestSelect
--- FAIL: TestSelect (0.00s)
	main_test.go:20: sqlite.Conn.Prepare: SQLITE_ERROR: no such table: test (SELECT * FROM test;)
FAIL
```

When run on a commit prior to c225a6cfce24bd7b14add5c91822cfe7a78c1bcb, I get the following output:

```plain
$ go test -v .
# crawshaw.io/sqlite
cgo-generated-wrappers:7:13: warning: built-in function ‘free’ declared as non-function
=== RUN   TestSelect
--- PASS: TestSelect (0.00s)
	main_test.go:23: []
PASS
```

--- a/sqlitex/pool.go
+++ b/sqlitex/pool.go
@@ -63,7 +63,7 @@ type Pool struct {
 //	SQLITE_OPEN_WAL
 //	SQLITE_OPEN_URI
 //	SQLITE_OPEN_NOMUTEX
-func Open(uri string, flags sqlite.OpenFlags, poolSize int) (*Pool, error) {
+func Open(uri string, flags sqlite.OpenFlags, poolSize int) (pool *Pool, err error) {
 	if uri == ":memory:" {
 		return nil, strerror{msg: `sqlite: ":memory:" does not work with multiple connections, use "file::memory:?mode=memory"`}
 	}
@@ -73,6 +73,12 @@ func Open(uri string, flags sqlite.OpenFlags, poolSize int) (*Pool, error) {
 		free:       make(chan *sqlite.Conn, poolSize),
 		closed:     make(chan struct{}),
 	}
+	defer func() {
+		// If an error occurred, call Close outside the lock so this doesn't deadlock.
+		if err != nil {
+			p.Close()
+		}
+	}()
 
 	if flags == 0 {
 		flags = sqlite.SQLITE_OPEN_READWRITE | sqlite.SQLITE_OPEN_CREATE | sqlite.SQLITE_OPEN_WAL | sqlite.SQLITE_OPEN_URI | sqlite.SQLITE_OPEN_NOMUTEX
@@ -88,7 +94,6 @@ func Open(uri string, flags sqlite.OpenFlags, poolSize int) (*Pool, error) {
 	for i := 0; i < poolSize; i++ {
 		conn, err := sqlite.OpenConn(uri, flags)
 		if err != nil {
-			p.Close()
 			return nil, err
 		}
 		p.free <- conn

--- a/sqlitex/pool.go
+++ b/sqlitex/pool.go
@@ -80,7 +80,7 @@ func Open(uri string, flags sqlite.OpenFlags, poolSize int) (*Pool, error) {
 
 	// sqlitex_pool is also defined in package sqlite
 	const sqlitex_pool = sqlite.OpenFlags(0x01000000)
-	flags &= sqlitex_pool
+	flags |= sqlitex_pool
 
 	p.allMu.Lock()
 	defer p.allMu.Unlock()

